lib/versioning/loose/index.ts

```diff
@@ -11,51 +11,50 @@ const versionPattern = regEx(/^v?(\d+(?:\.\d+)*)(.*)$/);
 const commitHashPattern = regEx(/^[a-f0-9]{7,40}$/);
 const numericPattern = regEx(/^[0-9]+$/);
 
-function parse(version: string): any {
-  if (commitHashPattern.test(version) && !numericPattern.test(version)) {
-    return null;
-  }
-  const matches = versionPattern.exec(version);
-  if (!matches) {
-    return null;
-  }
-  const [, prefix, suffix] = matches;
-  const release = prefix.split('.').map(Number);
-  if (release.length > 6) {
-    return null;
+class LooseVersioningApi extends GenericVersioningApi {
+  protected _parse(version: string): GenericVersion | null {
+    if (commitHashPattern.test(version) && !numericPattern.test(version)) {
+      return null;
+    }
+    const matches = versionPattern.exec(version);
+    if (!matches) {
+      return null;
+    }
+    const [, prefix, suffix] = matches;
+    const release = prefix.split('.').map(Number);
+    if (release.length > 6) {
+      return null;
+    }
+    return { release, suffix: suffix || '' };
   }
-  return { release, suffix: suffix || '' };
-}
 
-function compare(version1: string, version2: string): number {
-  const parsed1 = parse(version1);
-  const parsed2 = parse(version2);
-  // istanbul ignore if
-  if (!(parsed1 && parsed2)) {
-    return 1;
-  }
-  const length = Math.max(parsed1.release.length, parsed2.release.length);
-  for (let i = 0; i < length; i += 1) {
-    const part1 = parsed1.release[i];
-    const part2 = parsed2.release[i];
-    // shorter is smaller 2.1 < 2.1.0
-    if (part1 === undefined) {
-      return -1;
-    }
-    if (part2 === undefined) {
+  protected override _compare(version: string, other: string): number {
```

Comment:
I think we can completely remove this override, as the base makes exactly the same

Reply:
Implementations have some differences that cause tests to fail. If it's a problem, maybe let's address this with separate issue/PR?

Reply:
ok, but don't forget 😉